geth/peers/peerpool.go

```diff
@@ -185,7 +215,24 @@ func (p *PeerPool) handleServerPeers(server *p2p.Server, events <-chan *p2p.Peer
 			p.stopDiscovery(server)
 			return
 		case <-timeout:
-			log.Debug("DiscV5 timed out", "server", server.Self())
+			log.Info("DiscV5 timed out", "server", server.Self())
+
+			hasTopicBelowMin := false
```

Comment:
what was a reason to add it? i thought that hard-stop by timeout is a must, cause we will be draining battery/network, especially if discovery is in aggressive mode

Reply:
The aggressive mode also timeouts.

I am trying to experiment a bit. We talked about displaying a notification in the UI that Discovery V5 timed out and the user could resume it manually. But in such a case, when there are no peers, any other action than resuming Discovery does not make sense because the app without peers is useless.

Reply:
> I am trying to experiment a bit. We talked about displaying a notification in the UI that Discovery V5 timed out and the user could resume it manually. But in such a case, when there are no peers, any other action than resuming Discovery does not make sense because the app without peers is useless.

that's true, but i think if such event occurs it is better to ask a user for resume because there is clearly something wrong, either with the user internet or our cluster.

do you want to keep discovery in low mode all the time? the issue here is that all other peers that are also looking for peers will make any given device generate lots of responses.

Reply:
I am not sure what is the better option from the user perspective and how often this can happen. But yeah, resuming seems like a safer option as it will generate less traffic and allows us to get better feedback from the users (if displaying this notification is often, they will start to complain which tells us that there is a problem).

Reply:
It has been changed.